### ISSUE TITLE
Improve spatial join speed

### ISSUE THREAD

**Issue by EricSchles** [nycivicengagementtable/spatial_join#4]
Did you try just using fiona instead of geopandas?  Might be faster:

http://gis.stackexchange.com/questions/102933/more-efficient-spatial-join-in-python-without-qgis-arcgis-postgis-etc

Also, there is postgres-GIS (postgis):
http://workshops.boundlessgeo.com/postgis-intro/joins.html

it looks like heroku supports this:
https://devcenter.heroku.com/articles/heroku-postgres-extensions-postgis-full-text-search 
It's worth noting this is only available in the production tier so may not be a viable option.

I'll look into some further examples of speed ups.

Java may also be a viable option.

**Comment by EricSchles** [comment]
http://gis.stackexchange.com/questions/172144/fastest-way-to-join-many-points-to-many-polygons-in-python

seems to indicate you are already using the fastest method available in python...I'm going to look around a little more.

**Comment by EricSchles** [comment]
http://docs.paralleluniverse.co/spacebase/ - this looks like a possible avenue.  I'm going to try it out today.  They have a spatial join and a python client.

**Comment by afeld** [comment]
I was trying to avoid persisting anything, so skipped looking into
geo-friendly databases. Since I doubt the join time is never going to be
_fast_ for large files (like, in a reasonable time for an HTTP round-trip),
the medium-term way to go is probably to do the processing in the
background/asynchronously. That way, the time doesn't matter _as_ much.
Certainly would be great to speed it up though!

**Comment by afeld** [comment]
> Java may also be a viable option.

I would be very reluctant to add more complexity 😉

**Comment by EricSchles** [comment]
Yes, the async thing seems to be the way to go.  I'm going to look into async'ing this tomorrow.  I tried the database route and it was _extremely_ underwhelming.
